cli-plugins/manager/hooks.go

```diff
@@ -25,6 +29,9 @@ func RunPluginHooks(dockerCli command.Cli, rootCmd, subCommand *cobra.Command, p
 	subCmdName := subCommand.Name()
 	if plugin != "" {
 		subCmdName = plugin
+		if len(args) > 1 {
+			subCmdName += " " + args[1]
```

Comment:
Should we just include all args here, or would that also pass other command arguments?

Mostly thinking about commands nesting, like: ` docker buildx imagetools inspect` - with the current code, it wouldn't pass `inspect` right?

Reply:
Eep, you're definitely right. I guess there's no trivial solution here, and my concern is that we can't tell the args from the command without having full access to the command tree from the plugin, so we'll have to pass all of it.

From a trust perspective, in general I think that's okay since the plugin will only be invoked if configured to do so, but we also use this string `subCmdName` to match against what is in the config, which won't work if we include args.

A possible solution to that would be to instead of comparing directly, we can match the configuration as a prefix, which I'm happy with, but I want to benchmark the performance impact there.

WDYT?

Reply:
I'm wondering - could we just pass the common part between the invoked args and what plugin defined as hooks?

Then:
- if the plugin defined a hook on `build`, and user calls `docker build whatever`, we'd only pass `build`
- if the plugin defined a hook  `imagetools inspect`, and user calls `docker buildx imagetools inspect ubuntu`, we only pass `imagetools inspect`.

Reply:
Oh, that sounds like a good solution! I'll give it a try.

Reply:
> if the plugin defined a hook `imagetools`, and user calls `docker buildx imagetools inspect ubuntu`, we only pass `imagetools`.

@vvoland Seeing your example, wouldn't that potentially cause some issues if two plugins have a common subcommand name and want to invoke hooks on it?  Maybe i'm missing something

Eg. if we also had a hypothetical `docker utils imagetools inspect ubuntu`, shouldn't w have to pass `utils imagetools` instead of just `imagetools` for the two to be distinguishable?

Reply:
Unless I misunderstood, you'd still pass the plugin name, the plugin name is a part of the config that gets registered so there wouldn't be collisions there.

Reply:
Yeah I gave a bad example, but I meant what Laura understood, which is matching the prefix, and not any substring 😄